integration_tests/src/main/python/delta_lake_low_shuffle_merge_test.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pyspark.sql.functions as f
+import pytest
+
+from delta_lake_merge_common import *
+from marks import *
+from pyspark.sql.types import *
+from spark_session import *
+
+delta_merge_enabled_conf = copy_and_update(delta_writes_enabled_conf,
+                                           {"spark.rapids.sql.command.MergeIntoCommand": "true",
+                            "spark.rapids.sql.command.MergeIntoCommandEdge": "true",
+                            "spark.rapids.sql.delta.lowShuffleMerge.enabled": "true",
+                            "spark.rapids.sql.format.parquet.reader.type": "PERFILE"})
+
+@allow_non_gpu("RapidsRepartitionByFilePathExec", *delta_meta_allow)
+@delta_lake
+@ignore_order
+@pytest.mark.skipif(not ((is_databricks_runtime() and is_databricks133_or_later()) or
+                         (not is_databricks_runtime() and spark_version().startswith("3.4"))),
+                    reason="Delta Lake Low Shuffle Merge only supports Databricks 13.3 or OSS "
+                           "delta 2.4")
+@pytest.mark.parametrize("use_cdf", [True, False], ids=idfn)
+@pytest.mark.parametrize("num_slices", num_slices_to_test, ids=idfn)
+def test_delta_merge_when_fallback(spark_tmp_path, spark_tmp_table_factory, use_cdf,
+                                       num_slices):
+    # Need to eliminate duplicate keys in the source table otherwise update semantics are ambiguous
+    src_table_func = lambda spark: two_col_df(spark, int_gen, string_gen, num_slices=num_slices).groupBy("a").agg(f.max("b").alias("b"))
+    dest_table_func = lambda spark: two_col_df(spark, int_gen, string_gen, seed=1, num_slices=num_slices)
+    merge_sql = "MERGE INTO {dest_table} USING {src_table} ON {dest_table}.a == {src_table}.a" \
+                " WHEN MATCHED THEN UPDATE SET * WHEN NOT MATCHED THEN INSERT *"
+
+    conf = copy_and_update(delta_merge_enabled_conf,
+                           {"spark.rapids.sql.exec.RapidsRepartitionByFilePathExec": "false"})
```

Comment:
This conflates the original WHEN fallback test with another test checking for CPU fallback of scan or partition.  They need to be separate tests, otherwise we're not actually testing if we fallback to the CPU for a normal WHEN clause as expected, only testing that when a, probably never set in practice, unrelated config is also set.